spec/services/checksum_validator_spec.rb

```diff
@@ -526,4 +524,27 @@
       cv.validate_checksums
     end
   end
+
+  describe 'endpoint validation' do
+    it 'errors when endpoint is not an Endpoint object' do
+      cv = described_class.new(pres_copy, 'foo')
+      cv.valid?
+      expect(cv.errors.messages).to include(endpoint: ["must be an actual Endpoint"])
+    end
+    it 'errors when endpoint_type is not online' do
+      endpoint_type = EndpointType.create(type_name: 'foo', endpoint_class: 'archive')
```

Comment:
Why not use a factory?  We already have a special one for this case, like:
```ruby
let(:endpoint) { create(:archive_endpoint) }
...
  cv = described_class.new(pc, endpoint.endpoint_name)
```
Then you don't need to create `endpoint_type` directly at all.